.github/ISSUE_TEMPLATE/regression-report.yml

```diff
@@ -0,0 +1,91 @@
+name: ☣️ Regression
+description: Report a regression you've observed in a newer version of dbt
+title: "[Regression] <title>"
+labels: ["bug", "regression", "triage"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thanks for taking the time to fill out this regression report!
+  - type: checkboxes
+    attributes:
+      label: Is this a net-new regression in a recent version of dbt-core?
+      description: >
+        A regression is when documented functionality works as expected in an older version of dbt-core,
+        and no longer works after upgrading to a newer version of dbt-core
+      options:
+        - label: I believe this is a regression in dbt-core functionality
+          required: true
+        - label: I have searched the existing issues, and I could not find an existing issue for this regression
+          required: true
+  - type: textarea
+    attributes:
+      label: Current Behavior
+      description: A concise description of what you're experiencing.
+    validations:
+      required: false
+  - type: textarea
+    attributes:
+      label: Expected/Previous Behavior
+      description: A concise description of what you expected to happen.
+    validations:
+      required: false
+  - type: textarea
+    attributes:
+      label: Steps To Reproduce
+      description: Steps to reproduce the behavior.
+      placeholder: |
+        1. In this environment...
+        2. With this config...
+        3. Run '...'
+        4. See error...
+    validations:
+      required: false
+  - type: textarea
+    id: logs
+    attributes:
+      label: Relevant log output
+      description: |
+        If applicable, log output to help explain your problem.
+      render: shell
+    validations:
+      required: false
+  - type: textarea
```

Comment:
What if we ask for what version this functionality worked for them and then what version it broke in? Narrows down the window of possibility of where the bug was introduced for us